### PR TITLE
refactor(ivy): get rid of styling cleanup functions outside of styling code

### PR DESCRIPTION
**Pull request by matsko**
Prior to this patch, each time `advance()` would run (or when a
templateFn or hostBindings code exits) then the core change detection
code would check to see whether the styling data needs to be reset. This
patch removes that functionality and places everything inside of the
scheduled styling exit function. This means that each time one or more
styling bindings run (even if the value hasn't changed) then an exit
function will be scheduled and that will do all the cleanup.

### CODE REVIEW

File: packages/core/src/render3/styling_next/bindings.ts [hunk 1]
```diff
@@ -404,18 +404,22 @@ export function flushStyling(
     if (!isContextLocked(stylesContext, hostBindingsMode)) {
       lockAndFinalizeContext(stylesContext, hostBindingsMode);
     }
-    applyStylingViaContext(
-        stylesContext, renderer, element, data, state.stylesBitMask, setStyle, styleSanitizer,
-        hostBindingsMode);
+    if (state.stylesBitMask) {
```
Comment (mhevery) [inline packages/core/src/render3/styling_next/bindings.ts]: I wonder if writing it this way has a better performance?
```suggestion
    if (state.stylesBitMask !== 0) {
```

File: packages/core/src/render3/styling_next/bindings.ts [hunk 2]
```diff
@@ -404,18 +404,22 @@ export function flushStyling(
...
   }
 
   if (classesContext) {
     if (!isContextLocked(classesContext, hostBindingsMode)) {
       lockAndFinalizeContext(classesContext, hostBindingsMode);
     }
-    applyStylingViaContext(
-        classesContext, renderer, element, data, state.classesBitMask, setClass, null,
-        hostBindingsMode);
+    if (state.classesBitMask) {
```
Comment (mhevery) [inline packages/core/src/render3/styling_next/bindings.ts]: same here

**Review by mhevery — APPROVED** [review]
I ❤️ all of the removals

**Review by mhevery — APPROVED** [review]

File: aio/scripts/_payload-limits.json [hunk 3]
```diff
@@ -16,8 +16,8 @@
       "uncompressed": {
         "runtime-es5": 3042,
         "runtime-es2015": 3048,
-        "main-es5": 493330,
-        "main-es2015": 435296,
+        "main-es5": 492593,
+        "main-es2015": 438296,
```
Comment (gkalpak) [inline aio/scripts/_payload-limits.json]: Is this increase expected?

Comment (kara) [reply aio/scripts/_payload-limits.json]: It was discussed in the last PR. But the short answer is no, we don't know why, and he will look into that as the next PR.

**Comment by AndrewKushnir** [comment]
- [VE Presubmit](https://test.corp.google.com/ui#id=OCL:269407116:BASE:269407664:1568665802748:735a50f5)
- [Ivy Presubmit](https://test.corp.google.com/ui#id=OCL:269406554:BASE:269407279:1568665696477:6d9ae251)

**Pull request closed without merging**